### PR TITLE
feat: change API in favor of one data prop

### PR DESCRIPTION
**Pull request by frassinier**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/Talend/react-talend-forms/blob/master/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
Pass jsonSchema, uiSchema and formData objects to form

**What is the new behavior?**
Pass data object to form

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/Form.js [hunk 1]
```diff
@@ -28,14 +35,12 @@ class Form extends React.Component {
 	}
 
 	render() {
-		const schema = (this.props.data && this.props.data.jsonSchema) ||
-			this.props.jsonSchema || this.props.schema;
+		const schema = this.props.data && this.props.data.jsonSchema;
```
Comment (talend-glorieux) [inline src/Form.js]: Do we have a way to throw an error when the user doesn't pass at least a schema?

File: __tests__/Form.test.js [hunk 2]
```diff
@@ -34,17 +36,17 @@ describe('<Form/>', () => {
 	const onChange = jest.fn();
 
 	beforeEach(() => {
-		wrapper = mount(<Form schema={schema} onSubmit={onSubmit} onChange={onChange} />);
+		wrapper = mount(<Form data={data} onSubmit={onSubmit} onChange={onChange} />);
 	});
 
-	it('Renders the <FormControl/> component', () => {
+	xit('Renders the <FormControl/> component', () => {
```
Comment (talend-glorieux) [inline __tests__/Form.test.js]: Why are the test skipped?

File: src/templates/CustomFieldTemplate.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+/**
+ * Custom field template
+ * @param props
+ * @returns {*}
+ * @constructor
+ */
+const CustomFieldTemplate = props => {
+	const { classNames, help, description, errors, children } = props;
```
Comment (talend-glorieux) [inline src/templates/CustomFieldTemplate.js]: Is that assignation rely required? I'd rather have the spread done on the function arguments. Removes the allocations.

File: src/widgets/RadioOrSelectWidget.js [hunk 4]
```diff
@@ -16,11 +16,7 @@ const RadioOrSelectWidget = (props) => {
 	const enumOptions = options && options.enumOptions;
 	const shouldDisplayRadio = enumOptions && enumOptions.length <= MAX_TO_RADIO;
 
-	return (
-		<div>
-			{ shouldDisplayRadio ? <Radio {...props} /> : <Select {...props} /> }
-		</div>
-	);
+	return (shouldDisplayRadio ? <Radio {...props} /> : <Select {...props} />);
```
Comment (talend-glorieux) [inline src/widgets/RadioOrSelectWidget.js]: It doesn't seem like this warrants a ternary operation. Could you turn it into a simple if statement?

File: __tests__/Form.test.js (hunk 2 above)
Comment (frassinier) [inline __tests__/Form.test.js]: I'm wainting for https://github.com/mozilla-services/react-jsonschema-form/pull/366 and the form will no render react-bootstrap anymore

File: src/templates/CustomFieldTemplate.js (hunk 3 above)
Comment (frassinier) [inline src/templates/CustomFieldTemplate.js]: ok

File: __tests__/Form.test.js (hunk 2 above)
Comment (talend-glorieux) [inline __tests__/Form.test.js]: OK Makes sense :)

**Pull request merged**